Extend prevent-late-fallback by lock-counter (#1732)
rauc/rauc: docs/reference.rst

  It has no effect for ``plain`` bundles, as the signature verification already checks the
  whole bundle.

``prevent-late-fallback=<true/false>`` (deprecated, optional)
  .. note:: Values *<true/false>* for this option are deprecated. They will be removed in a future release
    The syntax has changed. Please use the new values.
    In order to get the same behaviour set ``prevent-late-fallback=mark-bad``.

> ``prevent-late-fallback=<disabled/mark-bad/lock-counter>`` (optional)

Reviewer: But in our case, no option is deprecated. The values are.

  In some use-cases, fallback to an older version must be prevented after the
  update is completed successfully ('rauc status mark-good' executed from the
  new version).

  Supported values are:

  * ``disable`` Default if not set. Do not prevent fallback to older version.
  * ``mark-bad`` If this option is enabled, RAUC will execute the equivalent of
    'rauc status mark-bad other' after marking the currently booted slot as good.
    This means that the other slot(s) is/are no longer eligible for fallback.
    It behaves exactly like the *deprecated* ``prevent-late-fallback=false``.

>   * ``lock-counter`` This is a slot independent setting that preserves

Reviewer: `prevent-late-fallback` is always slot-independent, no matter which value you give.

    the status of individual slots, allowing RAUC to maintain awareness of
    their respective states.
    The ``slots-locking`` state is stored in a separate variable that must be
    recognized by the ``bootloader``.
    Currently, this feature is only supported when ``bootloader``
    is set to ``barebox``, and the ``&bootstate`` in the devicetree must
    include a new variable, for example:

  .. code-block:: Devicetree

    bootstate {

            [...]

            attempts_locked@14 {
                    reg = <0x14 0x4>;
                    type = "uint32";
            };
    };

>   With this configuration, the bootloader is aware of the locking state due to the

Reviewer: ```suggestion
  With this configuration, barebox is aware of the locking state due to the
```

  usage of the variable in the devicetree. So the bootloader stops decrementing
  the ``remaining_attempts`` counter.

>   Slots locking is activated when 'rauc status mark-good' is executed

Reviewer: ```suggestion
  Boot counter locking is activated when 'rauc status mark-good' is executed
```

  and deactivated when 'rauc status mark-active' is executed. This can happen either
  implicitly during a RAUC update or explicitly through manual commands.

>   The current value of the ``slots_locked`` variable can be read via the commands

Reviewer: ```suggestion
  The current value of the ``attempts_locked`` state variable can be read via the commands
```

>   ``barebox-state -d`` or and ``rauc status``.

Reviewer: ```suggestion
  ``barebox-state -d`` and ``rauc status`` (without service).
```

>   It is not yet available via die D-Bus interface.

Reviewer: ```suggestion
  It is not yet available via the D-Bus interface and thus cannot be displayed when calling `rauc status` with the service enabled.
```

Author: nope wasn't intentional
